src/main/resources/assets/admin/common/js/form/ValidationRecording.ts

```diff
@@ -168,23 +168,14 @@ module api.form {
                 return false;
             }
 
-            const keys = mapA.keys();
-            let result: IteratorResult<string>;
-
-            do {
-                result = keys.next();
-
-                if (!result.done) {
-                    const key: string = result.value;
-
-                    if (!mapA.get(key).equals(mapB.get(key))) {
-                        return false;
-                    }
-                }
-
-            } while (!result.done);
+            const keys: string[] = [];
+            mapA.forEach((value: KeyBinding, key: string) => {
```

Comment:
Why is the type of value here KeyBinding? 
According to function signature it should be Equitable. I'm surprised TypeScript compiler does not complain.